### PR TITLE
QLSWG: Add KQL-M DSL Survey results from Sergey Mudrov

### PR DESCRIPTION
**Pull request by manolama**

### CODE REVIEW

File: working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
+# Kusto Query Language, Metrics Variant DSL Survey Response
+
+### **Short Description:**
+
+The Kusto Query Language (KQL-M) DSL is a variant of the Kusto Query Language (KQL) DSL that used in
```
Comment (vordoom) [inline working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md:5]: KQL-M stands for 'Kusto Query Language for Metric". Would suggest to modify text accordingly, i.e.:
The Kusto Query Language for Metrics (KQL-M) DSL is a variant...

File: working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+   scale performance efficiency was also one of the goals.
+
+1. **What deficiencies or omissions in other DSLs lead to your decision to design a new language?**:
+
+   The primary language of comparison was PromQL, thus one of the goals was to gain corresponding functional 
+   parity. Still, PromQL syntax was considered as a challenge for end users because it required extra learning. 
+   Thus the syntax was inherited from another DBMS language which is very popular internally (it is also exposed 
+   outside as Azure Data Explorer). Also worth mentioning is that PromQL philosophy of 'expression for a single 
+   data point' contradicted to existing data and execution model. Thus the new language expression was 
+   expected to define entire resultant data set (vectors shape).
```
Comment (yurishkuro) [inline working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md:24]: Not really clear what 'expression for a single data point' means here. Below you mention that KQL-M is not dealing with tabular data model, so afaiu your data model of a time series is a vector of <ts, number> tuples, with additional map of dimensions attached, same as in Prometheus.

Comment (manolama) [reply working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md:24]: I can't answer for Sergey but I thought it was referring to Prom's ability to return the last recorded data point, ala the `query` endpoint.

Comment (vjsamuel) [reply working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md:24]: can we create a git issue for this and follow up please?

File: working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+   Both.
+
+1. **Can users specify a source in the query language? E.g. a specific table or database?**:
+
+   Not yet, but there are no blockers to introduce it in the future (as long as the source data can be represented 
+   as time series).
+
+    1. **Can users join with other sources than the primary database? E.g. a CSV file, cloud databases, etc.?**:
+
+       Not yet.
```
Comment (vjsamuel) [inline working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md:57]: No vs Not Yet. does this mean that they have something in the works or is planned?

**Review by vjsamuel — APPROVED** [review]

File: working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md (hunk 2 above)
Comment (vordoom) [reply working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md:24]: Please correct me if I am wrong, but vector in PromQL semantics will be (unless it is a range vector):

<vm1; 5>
<vm2; 3>
<vm3; 6>
...
<vmN; X>

Our data format upon processing and output is somewhat like below:

<vm1; 5, 3, 1, 9, 2>
<vm2; 3, 2, 0, 3, 1>
<vm3; 6, , , 2, 5>
...
<vmN; X, . . . >

File: working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md (hunk 3 above)
Comment (vordoom) [reply working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md:57]: It just means there are no blockers.

**Comment by manolama** [comment]
Merging for now.

**Pull request merged**